### PR TITLE
Use self-contained pipeline run to import resources

### PR DESCRIPTION
**Pull request by eddycharly**
This pull request uses a self-contained pipeline run to import resources.
Fixing https://github.com/tektoncd/dashboard/issues/1257

It embeds the pipeline resource, pipeline spec and task spec.
This way it doesn't rely on externally created resources.

What was done:
- remove `pipeline0` and `pipeline0-task` resources
- modified `ImportResources` to build a self-contained pipeline run
- modified tests to reflect changes done in `ImportResources`

### CODE REVIEW

File: src/containers/ImportResources/ImportResources.js [hunk 1]
```diff
@@ -158,75 +152,187 @@ export class ImportResources extends Component {
...
+    const labels = getGitValues(repourl);
+
+    const payload = {
+      apiVersion: 'tekton.dev/v1beta1',
+      kind: 'PipelineRun',
+      metadata: {
+        name: `import-resources-${Date.now()}`,
+        labels: {
+          ...labels,
+          app: 'tekton-app'
```
[the remark this answers is not in the document]
Comment (a-roberts) [reply src/containers/ImportResources/ImportResources.js:302]: Yep, good idea and that label SGTM

**Comment by eddycharly** [comment]
@AlanGreene shall i rework the tests also ?

**Comment by AlanGreene** [comment]
It would be nice to separate the tests so we have 2 unit tests instead of the current one which is nearer to an integration test.
- ImportResources.test.js would mock the `API.importResources` call
- api/index.test.js would check that the PipelineRun spec is built correctly and `post`-ed to the expected URL

This would make it clearer that the underlying API(s) used is an implementation detail that the ImportResources component doesn't need to be aware of and make it easier to refactor the API layer in future without having to change the component tests.

**Comment by eddycharly** [comment]
@AlanGreene I made the changes and reworked the tests.

**Pull request merged**